### PR TITLE
Break out the contributing notes into CONTRIBUTING.md

### LINKED ISSUE #1398: contributing guidelines
it would be lovely to have a CONTRIBUTING.md document about rustup.

here's what GitHub's help has to say about it:
https://help.github.com/articles/setting-guidelines-for-repository-contributors/

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #1398

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+set when running your compiled `rustup-init` or the toolchains it installs, but _unset_
+when rebuilding `rustup` itself.
+
+We use `rustfmt` to keep our codebase consistently formatted.  Please ensure that
+you have correctly formatted your code (most editors will do this automatically
+when saving) or it may not pass the CI tests.
+
+Unless you explicitly state otherwise, any contribution intentionally
+submitted for inclusion in the work by you, as defined in the
+Apache-2.0 license, shall be dual licensed as above, without any
```
Comment (kinnison) [inline CONTRIBUTING.md]: I think `as above` should probably be `as in README.md` or somesuch

File: README.md [hunk 2]
```diff
@@ -32,7 +32,7 @@ And it runs on all platforms Rust supports, including Windows.
 * [Security](#security)
 * [FAQ](#faq)
 * [License](#license)
-* [Contributing](#contributing)
+* [Contributing](CONTRIBUTING.md).
```
Comment (kinnison) [inline README.md]: Why does this get a full-stop when the other bullets don't?

**Review by kinnison — COMMENTED** [review]
Fundamentally good, would approve with the tweaks to the wording in CONTRIBUTING.md, but would also like the bullets to be consistent in README.me :+1:

File: README.md (hunk 2 above)
Comment (dwijnand) [reply README.md]: Vim finger slip :)

**Review by kinnison — APPROVED** [review]
Thanks for that.

**Comment by nrc** [comment]
Thanks!

**Pull request merged**